lib/charliecloud.py

```diff
@@ -2143,6 +2143,24 @@ def request_raw(self, method, url, statuses, auth=None, **kwargs):
                   % (method, statuses, res.status_code, res.reason))
       except requests.exceptions.RequestException as x:
          FATAL("%s failed: %s" % (method, x))
+      # construct friendly Docker Hub info message
+      rl_msg = [] # rate limit message array
+      if ("ratelimit-limit" in res.headers):
+         rl_info = res.headers["ratelimit-limit"].split(";w=")
+         rl_msg.append("Rate Limit: %s pulls per %s hours" % (rl_info[0],
+                                                              int(rl_info[1]) // 3600))
+      if ("ratelimit-remaining" in res.headers):
+         remaining = res.headers["ratelimit-remaining"].split(";w=")[0]
+         rl_msg.append("Pulls Remaining: %s" % remaining)
+      if ("docker-ratelimit-source" in res.headers):
+         src = res.headers["docker-ratelimit-source"]
+         if (re.match(r"[0-9.a-f:]+", src).group(0) == src): # IP address
```

Comment:
1. The idiomatic way to do this comparison would be just whether the match object is None or not.
2. By convention, we use `re.search()` for everything, with `^` and/or `$` if needed.